kitsune/wiki/models.py

```diff
@@ -58,6 +61,34 @@ class _NotDocumentView(Exception):
     """A URL not pointing to the document view was passed to from_url()."""
 
 
+class DocumentManager(models.Manager):
+    def visible(self, user, **kwargs):
+        """
+        Documents are effectively invisible when they have no approved content,
+        and the given user is not a superuser, nor allowed to delete documents or
+        review revisions, nor a creator of one of the (yet unapproved) revisions.
+        """
+        locale = kwargs.get("locale")
+        qs = self.filter(**kwargs)
+        if user.is_authenticated:
+            if not (
+                user.is_superuser or can_delete_documents_or_review_revisions(user, locale=locale)
+            ):
+                # Authenticated users without permission to see documents that
+                # have no approved content, can only see those they have created.
+                rev_created_by_user = Revision.objects.filter(
+                    document=OuterRef("pk"), creator=user
+                )
+                qs = qs.filter(Q(current_revision__isnull=False) | Exists(rev_created_by_user))
+        else:
+            # Anonymous users only see documents with approved content.
+            qs = qs.filter(current_revision__isnull=False)
+        return qs
+
+    def get_visible(self, user, **kwargs):
```

Comment:
shouldn't we have more required args here to ensure a successful get operation?

Reply:
No, I don't think so. In this regard, `get_visible()` is  no different than Django's `get()` , which also does not ensure a successful operation in that either a `Model.DoesNotExist` or `Model.MultipleObjectsReturned` could be returned.